File: src/main/java/org/zenbot/szolnok/timetable/batch/dao/BusStopRepository.java
package org.zenbot.szolnok.timetable.batch.dao;

import org.springframework.data.mongodb.repository.MongoRepository;
import org.zenbot.szolnok.timetable.batch.domain.BusStopWithBuses;

import java.util.Optional;

public interface BusStopRepository extends MongoRepository<BusStopWithBuses, String> {

    Optional<BusStopWithBuses> findByBusStopName(String s);
}

File: szolnok-timetable-backend-batch-stops/src/main/kotlin/org/zenbot/szolnok/timetable/backend/batch/stops/dao/BusStopRepository.kt
package org.zenbot.szolnok.timetable.backend.batch.stops.dao

import org.springframework.data.mongodb.repository.MongoRepository
import org.zenbot.szolnok.timetable.backend.domain.document.stop.BusStopWithBuses
import java.util.Optional

interface BusStopRepository : MongoRepository<BusStopWithBuses, String> {

    fun findByBusStopName(s: String): Optional<BusStopWithBuses>
}
